tests/mattermost/main.tf

```diff
@@ -0,0 +1,74 @@
+ 
+variable "public_key" {
+  type = string
+}
+
+
+ 
+ 
+  terraform {
+    required_providers {
+      grid = {
+        source = "threefoldtech/grid"
+      }
+    }
+  }
+
+  provider "grid" {
+  }
+
+  # this data source is used to break circular dependency in cases similar to the following:
+  # vm: needs to know the domain in its init script
+  # gateway_name: needs the ip of the vm to use as backend.
+  # - the fqdn can be computed from grid_gateway_domain for the vm
+  # - the backend can reference the vm ip directly 
+  data "grid_gateway_domain" "domain" {
+    node = 2
```

Comment:
Change this node id with the id of a gateway node on testnet. same thing for the node id in line `59`